presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -457,34 +458,75 @@ std::shared_ptr<PrestoExchangeSource> PrestoExchangeSource::getSelfPtr() {
   return std::dynamic_pointer_cast<PrestoExchangeSource>(shared_from_this());
 }
 
+const PrestoExchangeSource::ConnectionPool&
+PrestoExchangeSource::ConnectionPools::get(
+    const proxygen::Endpoint& endpoint,
+    folly::IOThreadPoolExecutor* ioExecutor) {
+  return *pools_.withULockPtr([&](auto ulock) -> const ConnectionPool* {
+    auto it = ulock->find(endpoint);
+    if (it != ulock->end()) {
+      return &it->second;
+    }
+    auto wlock = ulock.moveFromUpgradeToWrite();
+    auto& pool = (*wlock)[endpoint];
+    if (!pool.eventBase) {
+      pool.eventBase = ioExecutor->getEventBase();
+      pool.sessionPool = std::make_unique<proxygen::SessionPool>(nullptr, 10);
+      // Creation of the timer is not thread safe, so we do it here instead of
+      // in the constructor of HttpClient.
+      pool.eventBase->timer();
+    }
+    return &pool;
+  });
+}
+
+void PrestoExchangeSource::ConnectionPools::destroy() {
+  pools_.withWLock([](auto& pools) {
+    for (auto& [_, pool] : pools) {
+      pool.eventBase->runInEventBaseThread(
+          [sessionPool = std::move(pool.sessionPool)] {});
+    }
```

Comment:
1. Why do we need to clear sessionPool objects in the Event Base Thread?

2. Does this 'runInEventBaseThread' run synchronously?